recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -172,11 +167,10 @@ def validate(self):
                 f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
             )
 
-        if self.settings.os != "Linux" and self.options.shared:
+        if Version(self.version) == "1.8.3" and self.settings.os != "Linux" and self.options.shared:
```

Comment:
This is the main change, where the shared build issue seems not to reproduce outside this version, and all 3 others compile fine for me locally on macos